src/tools/wasm-emscripten-finalize.cpp

```diff
@@ -180,16 +178,16 @@ int main(int argc, const char *argv[]) {
   }
   wasm.updateMaps();
 
-  if (isSideModule) {
+  if (isSharedLibrary) {
     generator.replaceStackPointerGlobal();
+    generator.generatePostInstantiateFunction();
   } else {
     generator.generateRuntimeFunctions();
     generator.generateMemoryGrowthFunction();
     generator.generateStackInitialization(initialStackPointer);
-    // emscripten calls this by default for side libraries so we only need
-    // to include in as a static ctor for main module case.
-    if (wasm.getExportOrNull("__post_instantiate")) {
-      initializerFunctions.push_back("__post_instantiate");
+    // For side modules this gets called via the __post_instantiate function
+    if (wasm.getExportOrNull("__wasm_call_ctors")) {
```

Comment:
Do we want to unify this `"__wasm_call_ctors"` string literal with the `WASM_CALL_CTORS` in wasm-emscripten.cpp? Alternatively, should we fold this inside wasm-emscripten.cpp?